modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -358,21 +358,18 @@ public TxManagerImpl(
                 new TxCleanupRequestSender(txMessageSender, placementDriverHelper, txStateVolatileStorage);
     }
 
-    // TODO: IGNITE-24363 - change action argument type to ReplicaGroupId.
     private CompletableFuture<Boolean> primaryReplicaEventListener(
             PrimaryReplicaEventParameters eventParameters,
-            Consumer<TablePartitionId> action
+            Consumer<ReplicationGroupId> action
     ) {
         return inBusyLock(busyLock, () -> {
             assertReplicationGroupType(eventParameters.groupId());
 
-            if (!(eventParameters.groupId() instanceof TablePartitionId)) {
+            if (!(eventParameters.groupId() instanceof TablePartitionId) && !(eventParameters.groupId() instanceof ZonePartitionId)) {
```

Comment:
Could you please add a todo with https://issues.apache.org/jira/browse/IGNITE-22522 in order not to forget to remove `!(eventParameters.groupId() instanceof TablePartitionId)` after we will fully switch to colocation?